smart-contract-usage-in-javascript-sdk.md

```diff
@@ -0,0 +1,326 @@
+const client = await Universal({
+    url: 'https://sdk-testnet.aepps.com', //replace with https://sdk-mainnet.aepps.com for mainnet
+    internalUrl: 'https://sdk-testnet.aepps.com', //replace with https://sdk-mainnet.aepps.com for mainnet
+    keypair: {
+        publicKey: 'REPLACE_WITH_PUBLIC_KEY',
+        secretKey: 'REPLACE_WITH_PRIVATE_KEY'
+    },
+    networkId: 'ae_uat', //replace with ae_mainnet for mainnet
+    nativeMode: true
+});
+```
+
+### 2. Reading the ExampleContract
+
+To read the contract code, add after the previously added lines:
+
+```
+// read the ExampleContract.aes from same directory
+const contract = fs.readFileSync('./ExampleContract.aes', 'utf-8');
+```
+
+### 3. Compiling the contract code
+
+To compile the contract code, add after the previously added lines:
+
+```
+// use the client to compile the contract code, catching eventual errors
+const compiled = await client.contractCompile(contract).catch(logError);
+
+// logging the bytecode output to the console to see the code work
+console.log('contract bytecode:', compiled.bytecode);
+```
+
+### 4. Deploying the contract bytecode to the blockchain
+
+To deploy the contract bytecode, add after the previously added lines:
+
+```
+// use the client to deploy the contract bytecode to the blockchain, catching eventual errors
+// we have to pass the initState where we choose 10000 as the number of available count and 15 as minimum price per item
+const deployed = await client.contractDeploy(compiled.bytecode, 'sophia', {initState: '(10000, 15)'}).catch(logError);
+
+// logging the contract adddress to the console, this is later used to call function of the contract
+console.log('deployed contract address:', deployed.address);
+```
+
+
+### 5. Calling the contract functions
+
+To call the `reserve` function of the deployed contract. It can be called by anyone using the blockchain. Add after the previously added lines:
+
+```
+// call the contract 'reserve' function using the deployed address, 'sophia-address' is used to indicate we are calling using the address
+// then we pass the arguments '(200)' as count that we want to reserve and as option the amount of tokens that should be added to the contract call
+// in our case 200 (the count) times 15 (the minimum price set)
+const calledReserve = await client.contractCall(deployed.address, 'sophia-address', deployed.address, 'reserve', {
+    args: '(200)',
+    options: {amount: 200 * 15}
+}).catch(decodeError);
+
+// we log if the contract call was successful, in this case if the success response is defined
+console.log('reservation successful:', !!calledReserve);
+```
+
+To static call the `available_count` function of the deployed contract, static calls are exectued on the connected node directly and don't require a transaction or any fee, but can only be used for functions that don't manipulate state. It can be called by anyone using the blockchain. Add after the previously added lines:
```

Comment:
"Aeternity blockchain supports static calls - calls that are exectued on the connected node directly and don't require a transaction or any fee. However, these static calls can only be used to call functions that do not manipulate the state. Lets use static call on the `available_count` function. Add the following block after the reserve code"